src/apify_client/clients/resource_clients/key_value_store.py

```diff
@@ -115,6 +140,71 @@ def get_record(self, key: str, *, as_bytes: bool = False, as_file: bool = False)
 
         return None
 
+    def get_record_as_bytes(self, key: str) -> Optional[Dict]:
+        """Retrieve the given record from the key-value store, without parsing it.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/record/get-record
+
+        Args:
+            key (str): Key of the record to retrieve
+
+        Returns:
+            dict, optional: The requested record, or None, if the record does not exist
+        """
+        try:
+            response = self.http_client.call(
+                url=self._url(f'records/{key}'),
+                method='GET',
+                params=self._params(),
+                parse_response=False,
+            )
+
+            return {
+                'key': key,
+                'value': response.content,
+                'content_type': response.headers['content-type'],
+            }
+
+        except ApifyApiError as exc:
+            _catch_not_found_or_throw(exc)
+
+        return None
+
+    @contextmanager
+    def stream_record(self, key: str) -> Iterator[Optional[Dict]]:
+        """Retrieve the given record from the key-value store, as a stream.
+
+        https://docs.apify.com/api/v2#/reference/key-value-stores/record/get-record
+
+        Args:
+            key (str): Key of the record to retrieve
+
+        Returns:
+            dict, optional: The requested record as a context-managed streaming Response, or None, if the record does not exist
+        """
+        response = None
+        try:
+            response = self.http_client.call(
+                url=self._url(f'records/{key}'),
+                method='GET',
+                params=self._params(),
+                parse_response=False,
+                stream=True,
+            )
+
+            yield {
+                'key': key,
+                'value': response,
+                'content_type': response.headers['content-type'],
+            }
+
+        except ApifyApiError as exc:
+            _catch_not_found_or_throw(exc)
+            yield None
```

Comment:
You don't have this catch for dataset streaming, is that intentional?

Reply:
Yup, that's intentional. This logic (catching in k-v store, not catching in dataset) is the same in the JS client, so when originally implementing the Python client, I just copied it over to keep it consistent.

The logic is not great:
- fetching a non-existent dataset returns None
- fetching a non-existent key-value store returns None
- fetching a non-existent key from an existent key-value store returns None
- fetching a non-existent key from a non-existent key-value store returns None
- fetching items from a non-existent dataset throws an error

The last two of those are inconsistent, but I think in a way they can make sense, if you think of key-value stores as folders with files, and datasets as tables.

Anyway, I would leave it like this, as to not make more breaking changes and logical differences from the JS client.